Process multiple assays

liz-fw opened the pull request:
issue #38

ErinCain reviewed (commented):
lgtm

ErinCain reviewed (approved):
looks great! Thanks

alvizek commented on R/extract-transform-sherlock.R:
```diff
@@ -158,28 +174,76 @@ process_plate_layout <- function(filepath, plate_size) {
+process_well_sample_details <- function(filepath, sample_type, layout_type, single_assay_type, plate_run_id) {
 
+  layout_type <- tolower(layout_type)
   sample_type_id <- if_else(sample_type == "mucus", 1, 2)
```
typo - `ifelse` rather than if_else

alvizek commented on R/extract-transform-sherlock.R:
```diff
@@ -158,28 +174,76 @@ process_plate_layout <- function(filepath, plate_size) {
-                        assay_type == "Ots16_Spring1" ~ 3,
-                        assay_type == "Ots16_Winter1" ~ 4)
-
   layout_raw <- suppressMessages(read_excel(filepath,
```
just a check - `readxl` package is included as a dependency.

alvizek commented on R/extract-transform-sherlock.R:
```diff
@@ -158,28 +174,76 @@ process_plate_layout <- function(filepath, plate_size) {
+  # split plate
+  if(str_detect(layout_type, "split_plate")) {
+
+    assay_ids <- case_when(layout_type == "split_plate_early_late" ~ c(1, 2),
```
maybe change `case_when` to `ifelse` because `ifelse` is faster and in base R.

alvizek commented on R/extract-transform-sherlock.R:
```diff
@@ -158,28 +174,76 @@ process_plate_layout <- function(filepath, plate_size) {
+    plate_layout <- layout_raw |>
+      pivot_longer(names_to="col_num", values_to = "sample_id", -...1) |>
+      rename(row_num = ...1) |>
+      mutate(assay_id = ifelse(col_num %in% 1:12, assay_ids[1], assay_ids[2])) |>
```
just a double check here that assay_id == 1 is "early" and assay_id == 2 is "late"

alvizek commented on R/extract-transform-sherlock.R:
```diff
@@ -158,28 +174,76 @@ process_plate_layout <- function(filepath, plate_size) {
+                           single_assay_type == "OTS28 Late" ~ 2,
+                           single_assay_type == "OTS16 Spring" ~ 3,
+                           single_assay_type == "OTS16 Winter" ~ 4)
+
```
could do something here (above) where convert to lowercase in case of typos?

alvizek reviewed (commented):
this goods to me too! i didn't run through the user_workflow but did test the function using the same file specified in the user_workflow.

liz-fw commented:
@alvizek i changed the ifelse statement and updated the single_assay_type arg to lower.  For one of the case_when statements, it wasn't assigning both assay ids so i kept it to case_when. Should be good to go!
